src/git.ts

```diff
@@ -0,0 +1,1264 @@
+
+    NEURO.client?.sendActionResult(actionData.id, true)
+
+    // Normalize the file path if provided; otherwise, use wildcard.
+    const revertFiles: string = filePath ? getNormalizedRepoPathForGit(filePath) : `*`;
+
+    // Compute an absolute path. If the removeFiles is already absolute, use it.
+    // Otherwise, join it with the repository's root path.
+    let absolutePath: string;
+    if (path.isAbsolute(revertFiles)) {
+        absolutePath = revertFiles;
+    } else {
+        absolutePath = path.join(repo.rootUri.fsPath, revertFiles);
+    }
+
+    // Pass the absolute path to the Git remove command.
+    repo.revert([absolutePath]).then(() => {
+        NEURO.client?.sendContext(`Removed ${revertFiles} from the index.`);
+    }, (err: string) => {
+        NEURO.client?.sendContext(`Reverting files failed`);
+        logOutput("ERROR", `Git revert failed: ${err}`)
+    });
+}
+
+export function handleGitCommit(actionData: any) {
+    if (!git) {
+        NEURO.client?.sendActionResult(actionData.id, true, 'Git extension not available.');
+        return;
+    }
+    if (!vscode.workspace.getConfiguration('neuropilot').get('permission.gitOperations', false)) {
+        NEURO.client?.sendActionResult(actionData.id, true, 'You do not have permission to perform Git operations.');
+        return;
+    }
+    if (!repo) {
+        NEURO.client?.sendActionResult(actionData.id, true, 'You are not in a git repository.');
+        return;
+    }
+
+    
+    const message = `Neuro commit: ${actionData.params?.message}`;
+    const commitOptions: string[] | undefined = actionData.params?.commitOptions;
+    let ExtraCommitOptions: CommitOptions | undefined = {}
+
+    if (!message) {
+        NEURO.client?.sendActionResult(actionData.id, false, "No commit message provided.")
+        return;
+    }
+    if (!commitOptions) {
+        ExtraCommitOptions = undefined
+    }
+    else {
+        let invalidCommitOptionCheck: boolean | undefined
+        let invalidCommitOptions: string[] = []
+        commitOptions.map((option) => {
+            if (!ExtraCommitOptions) return;
+            switch(option) {
+                case "amend":
+                    ExtraCommitOptions.amend = true
+                    break;
+                case "signoff":
+                    ExtraCommitOptions.signoff = true
+                    break;
+                case "verbose":
+                    ExtraCommitOptions.verbose = true
+                default:
+                    invalidCommitOptionCheck = true
+                    //invalidCommitOptions.
+            }
+        })
+        if (invalidCommitOptionCheck === true) {
+            NEURO.client?.sendActionResult(actionData.id, false, `Invalid commit options: ${invalidCommitOptions.map((option: string) => {option})}`)
+            return;
+        }
+    }
+
+    NEURO.client?.sendActionResult(actionData.id, true)
+
+    repo.inputBox.value = message;
+    repo.commit(message).then(() => {
```

Comment:
`ExtraCommitOptions` isn't being used.
```suggestion
    repo.commit(message, ExtraCommitOptions).then(() => {
```